pytorch_lightning/trainer/trainer.py

```diff
@@ -1383,16 +1383,22 @@ def __set_ckpt_path(self, ckpt_path: Optional[str], model_provided: bool, model_
             ckpt_path = "best"
 
         if ckpt_path == "best":
-            # if user requests the best checkpoint but we don't have it, error
+            if len(self.checkpoint_callbacks) > 1:
+                rank_zero_warn(
+                    f'`.{fn}(ckpt_path="best")` is called with Trainer configured with multiple `ModelCheckpoint`'
+                    " callbacks. It will use the best checkpoint path from first checkpoint callback."
+                )
+
             if not self.checkpoint_callback:
                 raise MisconfigurationException(
                     f'`.{fn}(ckpt_path="best")` is set but `ModelCheckpoint` is not configured.'
                 )
+
             if not self.checkpoint_callback.best_model_path:
                 if self.fast_dev_run:
                     raise MisconfigurationException(
-                        f"You cannot execute `.{fn}()` with `fast_dev_run=True` unless you do"
-                        f" `.{fn}(ckpt_path=PATH)` as no checkpoint path was generated during fitting."
+                        f'You cannot execute `.{fn}(ckpt_path="best")` with `fast_dev_run=True`.'
+                        f" an exact checkpoint path"
```

Comment:
I don't understand the relationship with this fix

Reply:
I know it's unrelated but just improved the msg here, nothing major.

Reply:
I don't think the message is clearer than it was before. "an exact checkpoint path" reads unfinished.

Reply:
updated!